dttools/src/jx_eval.c

```diff
@@ -342,13 +344,13 @@ static struct jx * jx_eval_operator( struct jx_operator *o, struct jx *context )
 	struct jx *result = NULL;
 
 	/*
-	For function calls select and project, convert the first argument
+	For function calls select, project, and like, convert the first argument
 	to a string as a way of deferring evaluation of expressions.
 	*/
 
 	if(o->type==JX_OP_CALL && jx_istype(o->left,JX_SYMBOL)) {
 		const char *name = o->left->u.symbol_name;
-		if(!strcmp("select",name) || !strcmp("project",name)) {
+		if(!strcmp("select",name) || !strcmp("project",name) || !strcmp("like",name)) {
```

Comment:
Is quoting the first argument necessary here? If I try to get the regex from a variable, it crashes

    expression: like(infile,"test")
    jx_function_like: jx_function.c:794[DEVELOPMENT]: Assertion 'jx_istype(val, JX_STRING)' failed.
    Aborted (core dumped)

Reply:
The quoting is necessary for the lazy eval of keys. I pushed a fix for using a variable as the regex: eval the regex onto the context before the matching. Give it a try now.